public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -262,7 +262,41 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
                 conceptService.searchConcepts(searchConceptsRequest.getQuery(), convertedConceptFilter,
                         searchConceptsRequest.getVocabularyIds(), domainIds, maxResults);
         ConceptListResponse response = new ConceptListResponse();
-        response.setItems(concepts.getContent().stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
+        List<Concept> matchedConcepts = concepts.getContent();
+        List<Concept> conceptCodeMatches = new ArrayList<>();
+        for(Concept con : matchedConcepts){
+            String conceptCode = con.getConceptCode();
+            Long conceptId = con.getConceptId();
+            if(!con.getStandardConcept().equals("S")){
+                if(conceptCode.equals(searchConceptsRequest.getQuery())) {
```

Comment:
This code and the next code are identical apart from the "matchType" right? I think you can simplify this code and just have:

response.setMatchType(conceptCode.equals(searchConceptsRequest.getQuery()) ? "ConceptCode" : "ConceptId")

and get rid of the if/else.

Reply:
Makes sense.